### PR TITLE
register/pmpaddrx: fix power of 2 check when napot encoding

### PR DESCRIPTION
**Pull request by beezow**
Fixes a bug when checking if a region can be encoded as NAPOT.

### CODE REVIEW

**Review by FawazTirmizi — APPROVED** [review]

File: src/register/pmpaddrx.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl PmpAddr {
         // verify size is not too big
         if (encoded_size > usize::MAX as Size) ||
             // check size is a power of 2
-            (size == (size & !(size-1))) ||
+            (size != (size & !(size-1))) ||
```
Comment (furquan-rivos) [inline src/register/pmpaddrx.rs]: Wouldn't it be easier to just check `((size & (size-1)) != 0)`? It effectively does the same thing, but I find the pattern easier to decode :rofl:

**Review by furquan-rivos — APPROVED** [review]

File: src/register/pmpaddrx.rs (hunk 1 above)
Comment (beezow) [reply src/register/pmpaddrx.rs]: Hm, I don't quite follow.  `((size & (size-1)) = size` iff size is a power of 2, so wouldn't `((size & (size-1)) != 0)` be true when size is a power of two?

Comment (furquan-rivos) [reply src/register/pmpaddrx.rs]: Consider size  = 4 (0b100)

size - 1 = 3 (0b011)

size & (size - 1)  = (0b100 & 0b011) = 0b000

So, size & (size - 1) returns 0 when size is a power of 2.

Comment (beezow) [reply src/register/pmpaddrx.rs]: Oh, I didn't notice you dropped the `!` before (size-1)!

Comment (beezow) [reply src/register/pmpaddrx.rs]: changed.

**Review by furquan-rivos — APPROVED** [review]

**Pull request merged**